modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts

```diff
@@ -134,17 +134,34 @@ const applyFormat = (ed: Editor, name: string, vars?: FormatVars, node?: Node |
       const process = (node: Node) => {
         let hasContentEditableState = false;
         let lastContentEditable = contentEditable;
+        let isWrappableNoneditableElm = false;
         const nodeName = node.nodeName.toLowerCase();
         const parentNode = node.parentNode as Node;
         const parentName = parentNode.nodeName.toLowerCase();
 
         // Node has a contentEditable value
-        if (NodeType.isElement(node) && dom.getContentEditable(node)) {
+        const contentEditableValue = dom.getContentEditable(node);
+        if (Type.isNonNullable(contentEditableValue)) {
           lastContentEditable = contentEditable;
-          contentEditable = dom.getContentEditable(node) === 'true';
-          hasContentEditableState = true; // We don't want to wrap the container only it's children
+          contentEditable = contentEditableValue === 'true';
+          // Unless the noneditable element is wrappable, we don't want to wrap the container, only it's editable children
+          hasContentEditableState = true;
+          isWrappableNoneditableElm = FormatUtils.isWrappableNoneditable(ed, node);
         }
 
+        const isMatchingWrappingBlock = isWrappingBlockFormat && MatchFormat.matchNode(ed, node, name, vars);
+
+        const isEditableDescendant = contentEditable && !hasContentEditableState;
+        const isValidBlockFormatForNode = isNonWrappingBlockFormat && FormatUtils.isTextBlock(ed, nodeName) && FormatUtils.isValid(ed, parentName, wrapName);
+        const canRenameBlock = isEditableDescendant && isValidBlockFormatForNode;
+
+        const isValidWrapNode = isWrapNameFormat && FormatUtils.isValid(ed, wrapName, nodeName) && FormatUtils.isValid(ed, parentName, wrapName);
+        // If it is not node specific, it means that it was not passed into 'formatter.apply` and is withiin the editor selection
+        const isZwsp = !nodeSpecific && NodeType.isText(node) && Zwsp.isZwsp(node.data);
+        const isCaret = isCaretNode(node);
+        const isCorrectFormatForNode = !FormatUtils.isInlineFormat(format) || !dom.isBlock(node);
+        const canWrapNode = (isEditableDescendant || isWrappableNoneditableElm) && isValidWrapNode && !isZwsp && !isCaret && isCorrectFormatForNode;
```

Comment:
This is doing a lot more eager evaluation now... can we break this out into functions that are called when needed to help with that? It's also help to make this more readable I believe.

Reply:
Thanks for the suggestion. I have set up some functions to avoid eager evaluation. Let me know if this still isn't quite what were you were thinking

Reply:
This still isn't quite what I was thinking but it's close... these are all still inline functions and I was hoping we could move them out so it's easier to read.

Reply:
I have moved them up. Thanks for pointing that out